src/LanguageServer/Protocol/Workspaces/LspWorkspaceManager.cs

```diff
@@ -210,8 +210,17 @@ async ValueTask TryCloseDocumentsInMutatingWorkspaceAsync(Uri uri)
             var registeredWorkspaces = _lspWorkspaceRegistrationService.GetAllRegistrations();
             foreach (var workspace in registeredWorkspaces)
             {
-                await ApplyChangeToMutatingWorkspaceAsync(workspace, uri, (_, documentId) =>
-                    workspace.TryOnDocumentClosedAsync(documentId, cancellationToken)).ConfigureAwait(false);
+                await ApplyChangeToMutatingWorkspaceAsync(workspace, uri, async (_, documentId) =>
+                {
+                    if (documentId.IsSourceGenerated)
+                    {
+                        // Source generated documents cannot go through OnDocumentOpened/Closed.
+                        // There is a separate OnSourceGeneratedDocumentOpened/Closed method, but there is no need
+                        // for us to call it in LSP - it deals with mapping TextBuffers to text containers.
+                        return;
```

Comment:
@jasonmalinowski @CyrusNajmabadi for thoughts

Reply:
i think the comment coudlo be beefed up.

Reply:
So I've forgotten, for _regular_ documents, why are we opening/closing the documents? Is it just so other code can ask "what's open" from the workspace or are we still getting some other benefit?

Reply:
I honestly can't remembner anything that would require the regular documents to actually be opened in the workspace.  Generally LSP features should be checking if LSP thinks the document is opened.

And there isn't any source text container stuff to do either, sicne we have our own LSP fork management.